data/src/test/java/com/linkedin/data/schema/resolver/TestDataSchemaResolver.java

```diff
@@ -460,22 +700,36 @@ public void testMapDataSchemaResolver()
                         "typeref has a circular reference to itself"
                     }
                 }
+        },
+        {
+            "Circular reference involving typerefs, using alias",
+            PDL,
+            asMap(buildSystemIndependentPath("a1", "typeref1.pdl"), "@aliases = [\"typerefAlias1\"] typeref typeref1 = typeref2",
+                  buildSystemIndependentPath("a1", "typeref2.pdl"), "typeref typeref2 = typerefAlias1"
+                 ),
+            new String[][]
+                {
+                    {
+                        "typeref1",
+                        ERROR,
+                        "typeref has a circular reference to itself"
+                    }
+                }
         }
     };
   }
 
   @Test(dataProvider = "circularReferenceData")
-  public void testCircularReferences(String desc, Map<String, String> testSchemas, String[][] testLookupAndExpectedResults)
+  public void testCircularReferences(String desc, SchemaFormatType extension, Map<String, String> testSchemas, String[][] testLookupAndExpectedResults)
   {
     boolean debug = false;
 
     for (String[] testLookupAndExpectedResult : testLookupAndExpectedResults)
     {
-      DataSchemaResolver resolver = new MapDataSchemaResolver(
```

Comment:
As I mentioned in my previous comment, using the enum gives you easy access to the schema parser instance. You shouldn't be doing `extension.equals(PDSC) ? SchemaParserFactory.instance() : PdlSchemaParserFactory.instance()`. You should be doing `extension.getSchemaParserFactory()`

Reply:
You could potentially also avoid having to pass in the schema parser factory into the `MapDataSchemaResolver` itself, you could just pass in only the extension, from which you could derive the parser factory and the string extension.

Reply:
I don't see how you second suggestion is significantly better, but I did the change according to your suggestion.